trade-finance-manager-api/src/v1/helpers/create-tasks.api-test.js

```diff
@@ -85,41 +85,69 @@ describe('defaults - tasks creation', () => {
       });
     });
 
-    describe('when excludedTasks array of strings is passed', () => {
-      it('should NOT return any tasks in a group that match a title in excludedTasks, retaining incremental task ids', () => {
+    describe('when no additionalTasks are passed and a task has isConditional flag', () => {
+      it('should NOT add isConditional tasks', () => {
         const mockGroupTasks = [
           { title: 'Task A', team: 'Test team' },
-          { title: 'Task B', team: 'Test team' },
+          { title: 'Task B', team: 'Test team', isConditional: true },
           { title: 'Task C', team: 'Test team' },
-          { title: 'Task D', team: 'Test team' },
+          { title: 'Task D', team: 'Test team', isConditional: true },
         ];
 
         const mockGroupId = 1;
 
-        const mockExcludedTasks = [
+        const result = createGroupTasks(
+          mockGroupTasks,
+          mockGroupId,
+        );
+
+        const conditionalTasks = mockGroupTasks.filter((task) => task.isConditional === true);
+
+        const conditionalTask1 = conditionalTasks[0];
+        const conditionalTask2 = conditionalTasks[1];
+
+        const conditionalTask1InResult = result.find((task) => task.title === conditionalTask1.title);
+        expect(conditionalTask1InResult).toBeUndefined();
+
+        const conditionalTask2InResult = result.find((task) => task.title === conditionalTask2.title);
+        expect(conditionalTask2InResult).toBeUndefined();
+      });
+    });
+
+    describe('when additionalTasks array of strings is passed', () => {
+      it('should return any additional tasks (with isConditional flag) in a group that match a title in additionalTasks, retaining incremental task ids', () => {
+        const mockGroupTasks = [
```

Comment:
could we rename this to something else? it looks like it was declared already on line 27

Reply:
Done here: https://github.com/UK-Export-Finance/dtfs2/pull/504/commits/8b46da370dbdc077d6e2441434661ffa529cf55c